src/tactics/FStar.Tactics.CtrlRewrite.fs

```diff
@@ -236,8 +236,9 @@ and on_subterms
 
       (* Descend on head and branches in parallel. Branches
        * are opened with their contexts extended. Ignore the when clause,
-       * and do not go into patterns. *)
-      | Tm_match (hd, brs) ->
+       * and do not go into patterns.
+       * also ignoring the return annotations *)
```

Comment:
Maybe we should descend into the returns? I guess skipping it is analogous to the treatment of Tm_ascribed